components/raftstore-v2/src/operation/command/write/raw_write.rs

```diff
@@ -0,0 +1,484 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::borrow::Cow;
+
+use engine_traits::{CF_DEFAULT, CF_LOCK, CF_WRITE};
+use kvproto::raft_cmdpb::{CmdType, RaftCmdRequest, RaftRequestHeader, Request};
+use protobuf::{CodedInputStream, Message, SingularPtrField};
+use tikv_util::Either;
+
+use crate::router::CmdResChannel;
+
+// MAGIC number to hint raw write is used.
+// TODO: use protobuf blob request seems better.
+const MAGIC_PREFIX: u8 = 0x00;
+
+/// We usually use `RaftCmdRequest` for read write request. But the codec is
+/// not efficient enough for simple request. `RawWrite` is introduce to make
+/// codec alloc less and fast.
+#[derive(Debug)]
+pub struct RawWriteEncoder {
```

Comment:
Is there any micro-benchmark of RawWriteEncoder vs RaftCmdRequest?

Reply:
No, but you can check the benchmark I did https://github.com/tikv/tikv/issues/12020.